src/common/snippets/tests/src/lowered/pass/buffer_allocation.cpp

```diff
@@ -0,0 +1,213 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "lowered/pass/buffer_allocation.hpp"
+
+#include "openvino/opsets/opset.hpp"
+#include "snippets/snippets_isa.hpp"
+#include "snippets/lowered/linear_ir.hpp"
+#include "snippets/lowered/pass/mark_loops.hpp"
+#include "snippets/lowered/pass/init_loops.hpp"
+#include "snippets/lowered/pass/insert_load_store.hpp"
+#include "snippets/lowered/pass/validate_loops.hpp"
+#include "snippets/lowered/pass/insert_loops.hpp"
+#include "snippets/lowered/pass/allocate_buffers.hpp"
+#include "snippets/lowered/pass/fuse_loops.hpp"
+#include "snippets/lowered/pass/split_loops.hpp"
+#include "snippets/lowered/pass/insert_buffers.hpp"
+#include "snippets/lowered/pass/softmax_decomposition.hpp"
+
+#include "common_test_utils/common_utils.hpp"
+
+
+namespace ov {
+namespace test {
+namespace snippets {
+
+std::string BufferAllocationTest::getTestCaseName(testing::TestParamInfo<ov::test::snippets::BufferAllocationParams> obj) {
+    bool is_optimized, with_split_loops;
+    size_t expected_size, expected_count;
+
+    std::tie(is_optimized, with_split_loops, expected_size, expected_count) = obj.param;
+
+    std::ostringstream result;
+    result << "Opt=" << ov::test::utils::bool2str(is_optimized) << "_";
+    result << "Split=" << ov::test::utils::bool2str(with_split_loops) << "_";
+    result << "ExpBufferSize=" << expected_size << "_";
+    result << "ExpBufferNum=" << expected_count;
+    return result.str();
+}
+
+void BufferAllocationTest::SetUp() {
+    bool is_optimized, with_split_loops;
+    std::tie(is_optimized, with_split_loops, m_expected_size, m_expected_count) = this->GetParam();
+
+    const auto body = GetModel();
+    m_linear_ir = ov::snippets::lowered::LinearIR(body, std::make_shared<ov::snippets::IShapeInferSnippetsFactory>());
+    m_linear_ir.set_loop_depth(m_loop_depth);
+    ApplyTransformations(is_optimized, with_split_loops);
+}
+
+void BufferAllocationTest::MarkOp(const std::shared_ptr<ov::Node>& node, const std::vector<size_t>& subtensor) {
+    for (const auto& input : node->inputs())
+        ov::snippets::lowered::PortDescriptorUtils::set_port_descriptor_ptr(
+            input, std::make_shared<ov::snippets::lowered::PortDescriptor>(input, subtensor));
+    for (const auto& output : node->outputs())
+        ov::snippets::lowered::PortDescriptorUtils::set_port_descriptor_ptr(
+            output, std::make_shared<ov::snippets::lowered::PortDescriptor>(output, subtensor));
+}
+
+void BufferAllocationTest::ApplyTransformations(bool is_optimized, bool with_split) {
```

Comment:
By any chance can we avoid transformation pipeline duplication with Snippets code? It just matter of maintenance cost

Reply:
I had to duplicate this code (almost completely) just to adjust transformation pipeline. For example, `SplitLoops` transformation changes a Linear IR so that it affects memory allocation algorithm. If `SplitLoops` is disabled, `Buffers` between `Brgemms` **can** reuse memory of `Buffers` before `Brgemm0` (because the lifetime of `Buffers` before `Brgemm0` ends with `Brgemm0` execution in this case). If `SplitLoops` is enabled, these `Buffers` between `Brgemms` **cannot** reuse memory of `Buffers` before `Brgemm0` (because the life time of `Buffers` before `Brgemm0`  ends  with splited (blocking) `Loop` execution). I'd like just to validate the both cases (they're the both important).

Unfortunately, At the moment we cannot flexibly enable or disable transformations (we don't have API for this) But seems like we can just implement pipeline config like in core manager. I've created the ticket '126214' for that.  I had to add parameter `with_split` to enable/disable  `SplitLoops` as temporary solution. 

I'd prefer to implement the feature in the another PR to avoid a lot of changes in one PR (this one is actually big). What do you think?